### ISSUE TITLE
Notebooks: PySpark3 has wrong syntax in 1st call  

### ISSUE THREAD

**Issue by kevcunnane** [Microsoft/azuredatastudio#4262]
Issue Type: <b>Bug</b>

Open a notebook from a SQL connection, and change to PySpark3.
- Add a new cell and paste in
```py
df = (spark.read.option("inferSchema", "true")
.option("header", "true")
.csv("/kctest/my.csv"))
df.show(10)
```
- Add a 2nd cell and paste the same content

Expected:
- No errors in either
Actual:
- Get intellisense errors in 1st cell only

![image](https://user-images.githubusercontent.com/10819925/53670436-973d0980-3c2f-11e9-8e3a-fa03f010bcb6.png)

I wonder is this something weird happening with SQL intellisense support. @YurongHe  could you look?

Azure Data Studio version: azuredatastudio 1.5.0 (9ca6200018fc206d67a47229f991901a8a453781, 2017-12-15T12:00:00.000Z)
OS version: Darwin x64 18.2.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz (8 x 2200)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 3|
|Memory (System)|16.00GB (0.68GB free)|
|Process Argv|. --remote-debugging-port=9222 --no-first-run --no-default-browser-check --inspect=5875 --no-cached-data|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
newdatabase|kev|0.0.3
server-report|Mic|0.1.5
queryplan-show|phi|0.0.2

</details>

**Comment by YurongHe** [comment]
\4\azuredatastudio\extensions\notebook\node_modules\@jupyterlab\services\lib\kernel\default.js:1428:25))", source:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js (2280)
[25572:0304/125652.116:INFO:CONSOLE(2280)] "%c[Extension Host] %c[Error - 12:56:52 PM] Request textDocument/definition failed. %c(atstomOutputChannel.appendLine (d:\4\azuredatastudio\extensions\mssql\out\main.js:266:17))", source: D:\4\azuredatastudio\node_modulesne.js\dist\zone-node.js (2280)
[25572:0304/125652.118:INFO:CONSOLE(2280)] "%c[Extension Host] %c  Message: System.ArgumentOutOfRangeException: Index was out of ran Must be non-negative and less than the size of the collection.Parameter name: index   at Babel.ParserGenerator.MinimalList`1.get_Item(Int32 index)
   at Babel.ParserGenerator.ModifiedMinimalList`1.get_Item(Int32 index)
   at Microsoft.SqlServer.Management.SqlParser.Parser.TokenManager.GetToken(Int32 tokenNumber)
   at Microsoft.SqlTools.ServiceLayer.LanguageServices.Completion.ScriptDocumentInfo.GetPeekDefinitionTokens(ScriptParseInfo scriptP
eInfo, Int32 startLine, Int32 startColumn) in D:\a\1\s\src\Microsoft.SqlTools.ServiceLayer\LanguageServices\ScriptDocumentInfo.cs:li158   at Microsoft.SqlTools.ServiceLayer.LanguageServices.LanguageService.GetDefinition(TextDocumentPosition textDocumentPosition, Scri
ile scriptFile, ConnectionInfo connInfo) in D:\a\1\s\src\Microsoft.SqlTools.ServiceLayer\LanguageServices\LanguageService.cs:line 12   at Microsoft.SqlTools.ServiceLayer.LanguageServices.LanguageService.HandleDefinitionRequest(TextDocumentPosition textDocumentPosin, RequestContext`1 requestContext) in D:\a\1\s\src\Microsoft.SqlTools.ServiceLayer\LanguageServices\LanguageService.cs:line 415  Code: 0  %c(at CustomOutputChannel.appendLine (d:\4\azuredatastudio\extensions\mssql\out\main.js:266:17))", source: D:\4\azuredatadio\node_modules\zone.js\dist\zone-node.js (2280)
[25572:0304/125853.285:INFO:CONSOLE(2280)] "%c WARN", source: D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js (2280)
[25572:0304/125853.285:INFO:CONSOLE(2280)] "%c WARN", source: D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js (2280)
[25572:0304/125853.286:INFO:CONSOLE(2280)] "%c WARN", source: D:\4\azuredatastudio\node_modules\zone.js\dist\zone-node.js (2280)

**Comment by kevcunnane** [comment]
This looks like intellisense errors where it's requesting intellisense for the doc, but it's no longer registered in the SqlToolsService? Probably some issue with how we're changing language mode for the cell vs. still having a connection registered?

**Comment by kevcunnane** [comment]
This should be solved by your fix / the larger fix. We're attaching to the SQL client session and don't get a chance to respond to Jupyter kernel change events by updating language mode. Once we have a single activeClientSession, we at least have a chance to get this sorted...

**Comment by YurongHe** [comment]
Fixed by #4371
